python/python/lancedb/index.py

```diff
@@ -108,7 +132,29 @@ def __init__(
 
 
 class HnswSq:
-    """Describe a HNSW-SQ index configuration."""
+    """Describe a HNSW-SQ index configuration.
+    
+    HNSW-SQ stands for Hierarchical Navigable Small World - Scalar Quantization.
+    It is a variant of the HNSW algorithm that uses scalar quantization to compress
+    the vectors. To create an HNSW-SQ index, you can specify the following parameters:
+    
+    1. distance_type: str
+        The distance metric used to train the index. The following distance types are available:
+        - "l2" - Euclidean distance
+        - "cosine" - Cosine distance
+        - "dot" - Dot product
+        - "hamming" - Hamming distance
+    2. num_partitions: int
+        Indicates the number of partitions to divide the data into.
+    3. max_iterations: int
+        Max iterations.
+    4. sample_rate: int
+        Defines the rate at which samples are taken from the data.
+    5. m: int
+        It represents the number of edges per node in a graph
+    6. ef_construction: int
+        The number of neighbors to search for each vector during the construction phase.
```

Comment:
Please document in the `__init__`. Notice how in the `IvfPq` class we use the format:

```text
        Parameters
        ----------
        distance_type: str, default "L2"
            description...
```

This is the Numpy doc style, which you can read more about here: https://numpydoc.readthedocs.io/en/latest/format.html#parameters